composite_demo文件夹中client.py量化(int4)配置了但是显存没有减少还是12710MiB

### System Info / 系統信息

感谢开源贡献，自己在尝试用的时候使用的conda环境，都运行起来了，但是(int4)配置了但是显存没有减少还是12710MiB 
# packages in environment at /home/work/miniconda3/envs/ChatGLM3cuda121:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
accelerate                0.28.0                   pypi_0    pypi
aiofiles                  23.2.1                   pypi_0    pypi
aiohttp                   3.9.3                    pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
altair                    5.2.0                    pypi_0    pypi
annotated-types           0.6.0                    pypi_0    pypi
anyio                     4.3.0                    pypi_0    pypi
arxiv                     2.1.0                    pypi_0    pypi
asttokens                 2.4.1                    pypi_0    pypi
async-timeout             4.0.3                    pypi_0    pypi
attrs                     23.2.0                   pypi_0    pypi
blas                      1.0                         mkl    conda-forge
blinker                   1.7.0                    pypi_0    pypi
brotli-python             1.1.0           py310hc6cd4ac_1    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
cachetools                5.3.3                    pypi_0    pypi
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
click                     8.1.7                    pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
comm                      0.2.2                    pypi_0    pypi
contourpy                 1.2.0                    pypi_0    pypi
cpm-kernels               1.0.11                   pypi_0    pypi
cuda                      12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-cccl                 12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-command-line-tools   12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-compiler             12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-cudart               12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-cudart-dev           12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-cudart-static        12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-cuobjdump            12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-cupti                12.1.62                       0    nvidia/label/cuda-12.1.0
cuda-cupti-static         12.1.62                       0    nvidia/label/cuda-12.1.0
cuda-cuxxfilt             12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-demo-suite           12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-documentation        12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-driver-dev           12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-gdb                  12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-libraries            12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-libraries-dev        12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-libraries-static     12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-nsight               12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-nsight-compute       12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-nvcc                 12.1.66                       0    nvidia/label/cuda-12.1.0
cuda-nvdisasm             12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-nvml-dev             12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-nvprof               12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-nvprune              12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-nvrtc                12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-nvrtc-dev            12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-nvrtc-static         12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-nvtx                 12.1.66                       0    nvidia/label/cuda-12.1.0
cuda-nvvp                 12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-opencl               12.1.56                       0    nvidia/label/cuda-12.1.0
cuda-opencl-dev           12.1.56                       0    nvidia/label/cuda-12.1.0
cuda-profiler-api         12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-runtime              12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-sanitizer-api        12.1.55                       0    nvidia/label/cuda-12.1.0
cuda-toolkit              12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-tools                12.1.0                        0    nvidia/label/cuda-12.1.0
cuda-visual-tools         12.1.0                        0    nvidia/label/cuda-12.1.0
cycler                    0.12.1                   pypi_0    pypi
dataclasses-json          0.6.4                    pypi_0    pypi
debugpy                   1.8.1                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
distro                    1.9.0                    pypi_0    pypi
exceptiongroup            1.2.0                    pypi_0    pypi
executing                 2.0.1                    pypi_0    pypi
fastapi                   0.110.0                  pypi_0    pypi
feedparser                6.0.10                   pypi_0    pypi
ffmpeg                    4.3                  hf484d3e_0    pytorch
ffmpy                     0.3.2                    pypi_0    pypi
filelock                  3.13.3             pyhd8ed1ab_0    conda-forge
fonttools                 4.50.0                   pypi_0    pypi
freetype                  2.12.1               h267a509_2    conda-forge
frozenlist                1.4.1                    pypi_0    pypi
fsspec                    2024.3.1                 pypi_0    pypi
gds-tools                 1.6.0.25                      0    nvidia/label/cuda-12.1.0
gitdb                     4.0.11                   pypi_0    pypi
gitpython                 3.1.42                   pypi_0    pypi
gmp                       6.3.0                h59595ed_1    conda-forge
gmpy2                     2.1.2           py310h3ec546c_1    conda-forge
gnutls                    3.6.13               h85f3911_1    conda-forge
gradio                    4.24.0                   pypi_0    pypi
gradio-client             0.14.0                   pypi_0    pypi
greenlet                  3.0.3                    pypi_0    pypi
h11                       0.14.0                   pypi_0    pypi
httpcore                  1.0.5                    pypi_0    pypi
httpx                     0.27.0                   pypi_0    pypi
huggingface-hub           0.20.3                   pypi_0    pypi
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.6                pyhd8ed1ab_0    conda-forge
importlib-resources       6.4.0                    pypi_0    pypi
intel-openmp              2023.1.0         hdb19cb5_46306    defaults
ipykernel                 6.29.4                   pypi_0    pypi
ipython                   8.22.2                   pypi_0    pypi
jedi                      0.19.1                   pypi_0    pypi
jinja2                    3.1.3              pyhd8ed1ab_0    conda-forge
joblib                    1.3.2                    pypi_0    pypi
jpeg                      9e                   h166bdaf_2    conda-forge
jsonpatch                 1.33                     pypi_0    pypi
jsonpointer               2.4                      pypi_0    pypi
jsonschema                4.21.1                   pypi_0    pypi
jsonschema-specifications 2023.12.1                pypi_0    pypi
jupyter-client            8.6.1                    pypi_0    pypi
jupyter-core              5.7.2                    pypi_0    pypi
kiwisolver                1.4.5                    pypi_0    pypi
lame                      3.100             h166bdaf_1003    conda-forge
langchain                 0.1.13                   pypi_0    pypi
langchain-community       0.0.29                   pypi_0    pypi
langchain-core            0.1.35                   pypi_0    pypi
langchain-text-splitters  0.0.1                    pypi_0    pypi
langchainhub              0.1.15                   pypi_0    pypi
langsmith                 0.1.36                   pypi_0    pypi
latex2mathml              3.77.0                   pypi_0    pypi
lcms2                     2.15                 hfd0df8a_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libblas                   3.9.0           1_h86c2bf4_netlib    conda-forge
libcblas                  3.9.0           5_h92ddd45_netlib    conda-forge
libcublas                 12.1.0.26                     0    nvidia/label/cuda-12.1.0
libcublas-dev             12.1.0.26                     0    nvidia/label/cuda-12.1.0
libcublas-static          12.1.0.26                     0    nvidia/label/cuda-12.1.0
libcufft                  11.0.2.4                      0    nvidia/label/cuda-12.1.0
libcufft-dev              11.0.2.4                      0    nvidia/label/cuda-12.1.0
libcufft-static           11.0.2.4                      0    nvidia/label/cuda-12.1.0
libcufile                 1.6.0.25                      0    nvidia/label/cuda-12.1.0
libcufile-dev             1.6.0.25                      0    nvidia/label/cuda-12.1.0
libcufile-static          1.6.0.25                      0    nvidia/label/cuda-12.1.0
libcurand                 10.3.2.56                     0    nvidia/label/cuda-12.1.0
libcurand-dev             10.3.2.56                     0    nvidia/label/cuda-12.1.0
libcurand-static          10.3.2.56                     0    nvidia/label/cuda-12.1.0
libcusolver               11.4.4.55                     0    nvidia/label/cuda-12.1.0
libcusolver-dev           11.4.4.55                     0    nvidia/label/cuda-12.1.0
libcusolver-static        11.4.4.55                     0    nvidia/label/cuda-12.1.0
libcusparse               12.0.2.55                     0    nvidia/label/cuda-12.1.0
libcusparse-dev           12.0.2.55                     0    nvidia/label/cuda-12.1.0
libcusparse-static        12.0.2.55                     0    nvidia/label/cuda-12.1.0
libdeflate                1.17                 h0b41bf4_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_5    conda-forge
libgfortran-ng            13.2.0               h69a702a_5    conda-forge
libgfortran5              13.2.0               ha4646dd_5    conda-forge
libgomp                   13.2.0               h807b86a_5    conda-forge
libhwloc                  2.9.3           default_h554bfaf_1009    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libjpeg-turbo             2.0.0                h9bf148f_0    pytorch
liblapack                 3.9.0           5_h92ddd45_netlib    conda-forge
libnpp                    12.0.2.50                     0    nvidia/label/cuda-12.1.0
libnpp-dev                12.0.2.50                     0    nvidia/label/cuda-12.1.0
libnpp-static             12.0.2.50                     0    nvidia/label/cuda-12.1.0
libnsl                    2.0.1                hd590300_0    conda-forge
libnvjitlink              12.1.105                      0    nvidia
libnvjitlink-dev          12.1.55                       0    nvidia/label/cuda-12.1.0
libnvjpeg                 12.1.0.39                     0    nvidia/label/cuda-12.1.0
libnvjpeg-dev             12.1.0.39                     0    nvidia/label/cuda-12.1.0
libnvjpeg-static          12.1.0.39                     0    nvidia/label/cuda-12.1.0
libnvvm-samples           12.1.55                       0    nvidia/label/cuda-12.1.0
libpng                    1.6.43               h2797004_0    conda-forge
libsqlite                 3.45.2               h2797004_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_5    conda-forge
libtiff                   4.5.0                h6adf6a1_2    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libwebp-base              1.3.2                hd590300_0    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libxml2                   2.12.6               h232c23b_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
llvm-openmp               15.0.7               h0cdce71_0    conda-forge
loguru                    0.7.2                    pypi_0    pypi
markdown                  3.6                      pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                2.1.5           py310h2372a71_0    conda-forge
marshmallow               3.21.1                   pypi_0    pypi
matplotlib                3.8.3                    pypi_0    pypi
matplotlib-inline         0.1.6                    pypi_0    pypi
mdtex2html                1.3.0                    pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
mkl                       2023.1.0         h213fc3f_46344    defaults
mpc                       1.3.1                hfe3b2da_0    conda-forge
mpfr                      4.2.1                h9458935_0    conda-forge
mpmath                    1.3.0              pyhd8ed1ab_0    conda-forge
multidict                 6.0.5                    pypi_0    pypi
mypy-extensions           1.0.0                    pypi_0    pypi
ncurses                   6.4.20240210         h59595ed_0    conda-forge
nest-asyncio              1.6.0                    pypi_0    pypi
nettle                    3.6                  he412f7d_0    conda-forge
networkx                  3.2.1              pyhd8ed1ab_0    conda-forge
nsight-compute            2023.1.0.15                   0    nvidia/label/cuda-12.1.0
numpy                     1.26.4          py310hb13e2d6_0    conda-forge
openai                    1.14.3                   pypi_0    pypi
openh264                  2.1.1                h780b84a_0    conda-forge
openjpeg                  2.5.0                hfec8fc6_2    conda-forge
openssl                   3.2.1                hd590300_1    conda-forge
orjson                    3.10.0                   pypi_0    pypi
packaging                 23.2                     pypi_0    pypi
pandas                    2.2.1                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
peft                      0.10.0                   pypi_0    pypi
pexpect                   4.9.0                    pypi_0    pypi
pillow                    10.2.0                   pypi_0    pypi
pip                       24.0               pyhd8ed1ab_0    conda-forge
platformdirs              4.2.0                    pypi_0    pypi
prompt-toolkit            3.0.43                   pypi_0    pypi
protobuf                  4.25.3                   pypi_0    pypi
psutil                    5.9.8                    pypi_0    pypi
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
pyarrow                   15.0.2                   pypi_0    pypi
pydantic                  2.6.4                    pypi_0    pypi
pydantic-core             2.16.3                   pypi_0    pypi
pydeck                    0.8.1b0                  pypi_0    pypi
pydub                     0.25.1                   pypi_0    pypi
pygments                  2.17.2                   pypi_0    pypi
pyjwt                     2.8.0                    pypi_0    pypi
pyparsing                 3.1.2                    pypi_0    pypi
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.10.0          h543edf9_3_cpython    conda-forge
python-dateutil           2.9.0.post0              pypi_0    pypi
python-multipart          0.0.9                    pypi_0    pypi
python_abi                3.10                    4_cp310    conda-forge
pytorch                   2.1.2           py3.10_cuda12.1_cudnn8.9.2_0    pytorch
pytorch-cuda              12.1                 ha16c6d3_5    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2024.1                   pypi_0    pypi
pyyaml                    6.0.1           py310h2372a71_1    conda-forge
pyzmq                     25.1.2                   pypi_0    pypi
readline                  8.2                  h8228510_1    conda-forge
referencing               0.34.0                   pypi_0    pypi
regex                     2023.12.25               pypi_0    pypi
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
rich                      13.7.1                   pypi_0    pypi
rpds-py                   0.18.0                   pypi_0    pypi
ruff                      0.3.4                    pypi_0    pypi
safetensors               0.4.2                    pypi_0    pypi
scikit-learn              1.4.1.post1              pypi_0    pypi
scipy                     1.12.0                   pypi_0    pypi
semantic-version          2.10.0                   pypi_0    pypi
sentence-transformers     2.6.1                    pypi_0    pypi
sentencepiece             0.2.0                    pypi_0    pypi
setuptools                69.2.0             pyhd8ed1ab_0    conda-forge
sgmllib3k                 1.0.0                    pypi_0    pypi
shellingham               1.5.4                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
smmap                     5.0.1                    pypi_0    pypi
sniffio                   1.3.1                    pypi_0    pypi
sqlalchemy                2.0.29                   pypi_0    pypi
sqlite                    3.45.2               h2c6b66d_0    conda-forge
sse-starlette             2.0.0                    pypi_0    pypi
stack-data                0.6.3                    pypi_0    pypi
starlette                 0.36.3                   pypi_0    pypi
streamlit                 1.32.2                   pypi_0    pypi
sympy                     1.12            pypyh9d50eac_103    conda-forge
tbb                       2021.11.0            h00ab1b0_1    conda-forge
tenacity                  8.2.3                    pypi_0    pypi
threadpoolctl             3.4.0                    pypi_0    pypi
tiktoken                  0.6.0                    pypi_0    pypi
timm                      0.9.16                   pypi_0    pypi
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tokenizers                0.15.2                   pypi_0    pypi
toml                      0.10.2                   pypi_0    pypi
tomlkit                   0.12.0                   pypi_0    pypi
toolz                     0.12.1                   pypi_0    pypi
torchaudio                2.1.2               py310_cu121    pytorch
torchtriton               2.1.0                     py310    pytorch
torchvision               0.16.2              py310_cu121    pytorch
tornado                   6.4                      pypi_0    pypi
tqdm                      4.66.2                   pypi_0    pypi
traitlets                 5.14.2                   pypi_0    pypi
transformers              4.39.1                   pypi_0    pypi
typer                     0.11.0                   pypi_0    pypi
types-requests            2.31.0.20240311          pypi_0    pypi
typing-inspect            0.9.0                    pypi_0    pypi
typing_extensions         4.10.0             pyha770c72_0    conda-forge
tzdata                    2024.1                   pypi_0    pypi
urllib3                   2.2.1              pyhd8ed1ab_0    conda-forge
uvicorn                   0.29.0                   pypi_0    pypi
watchdog                  4.0.0                    pypi_0    pypi
wcwidth                   0.2.13                   pypi_0    pypi
websockets                11.0.3                   pypi_0    pypi
wheel                     0.43.0             pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.11               hd590300_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
yarl                      1.9.4                    pypi_0    pypi
zhipuai                   2.0.1                    pypi_0    pypi
zlib                      1.2.13               hd590300_5    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge


### Who can help? / 谁可以帮助到您？

_No response_

### Information / 问题信息

- [X] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

自己尝试了 add .quantize(4).cuda() before .eval() and remove device_map="auto" to use int4 model   直接报错了 
`Traceback (most recent call last):
  File "/home/work/miniconda3/envs/ChatGLM3cuda121/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 542, in _run_script
    exec(code, module.__dict__)
  File "/data/workspace/condaProject/ChatGLM3-main/composite_demo/main.py", line 10, in <module>
    import demo_chat, demo_ci, demo_tool
  File "/data/workspace/condaProject/ChatGLM3-main/composite_demo/demo_chat.py", line 7, in <module>
    client = get_client()
  File "/home/work/miniconda3/envs/ChatGLM3cuda121/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 210, in wrapper
    return cached_func(*args, **kwargs)
  File "/home/work/miniconda3/envs/ChatGLM3cuda121/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 239, in __call__
    return self._get_or_create_cached_value(args, kwargs)
  File "/home/work/miniconda3/envs/ChatGLM3cuda121/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 266, in _get_or_create_cached_value
    return self._handle_cache_miss(cache, value_key, func_args, func_kwargs)
  File "/home/work/miniconda3/envs/ChatGLM3cuda121/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 322, in _handle_cache_miss
    computed_value = self._info.func(*func_args, **func_kwargs)
  File "/data/workspace/condaProject/ChatGLM3-main/composite_demo/client.py", line 27, in get_client
    client = HFClient(MODEL_PATH, TOKENIZER_PATH, PT_PATH)
  File "/data/workspace/condaProject/ChatGLM3-main/composite_demo/client.py", line 152, in __init__
    self.model = AutoModel.from_pretrained(MODEL_PATH, trust_remote_code=True).quantize(4).cuda().eval()
  File "/home/work/.cache/huggingface/modules/transformers_modules/a5ba5501eb873d40d48bd0983bd2a8dd006bb838/modeling_chatglm.py", line 1212, in quantize
    self.transformer.encoder = quantize(self.transformer.encoder, bits, empty_init=empty_init, device=device,
  File "/home/work/.cache/huggingface/modules/transformers_modules/a5ba5501eb873d40d48bd0983bd2a8dd006bb838/quantization.py", line 156, in quantize
    layer.self_attention.query_key_value = QuantizedLinear(
  File "/home/work/.cache/huggingface/modules/transformers_modules/a5ba5501eb873d40d48bd0983bd2a8dd006bb838/quantization.py", line 128, in __init__
    assert str(weight.device).startswith('cuda'), 'The weights that need to be quantified should be on the CUDA device'
AssertionError: The weights that need to be quantified should be on the CUDA device
`
然后, device_map="auto" 或者  device_map="cuda:0"  是可以运行的但是显存没有减少，调用如下

            self.model = AutoModel.from_pretrained(
                model_path,
                trust_remote_code=True,
                #config=config, device_map="auto").eval()
                config=config, device_map="cuda:0").quantize(4).cuda().eval()
            # add .quantize(4).cuda() before .eval() and remove device_map="auto" to use int4 model

### Expected behavior / 期待表现

希望帮助确认具体的原因~

各种操作都试验了 不是显存占12G就是加载报错，麻烦帮确认下~
1.   device_map="cuda:0"删了  就是 **self.model = AutoModel.from_pretrained(MODEL_PATH, trust_remote_code=True).quantize(4).cuda().eval()**  会报错  
    client = HFClient(MODEL_PATH, TOKENIZER_PATH, PT_PATH)
  File "/data/workspace/condaProject/ChatGLM3-main/composite_demo/client.py", line 156, in __init__
    self.model = AutoModel.from_pretrained(MODEL_PATH, trust_remote_code=True).quantize(4).cuda().eval()
  File "/home/work/.cache/huggingface/modules/transformers_modules/a5ba5501eb873d40d48bd0983bd2a8dd006bb838/modeling_chatglm.py", line 1212, in quantize
    self.transformer.encoder = quantize(self.transformer.encoder, bits, empty_init=empty_init, device=device,
  File "/home/work/.cache/huggingface/modules/transformers_modules/a5ba5501eb873d40d48bd0983bd2a8dd006bb838/quantization.py", line 156, in quantize
    layer.self_attention.query_key_value = QuantizedLinear(
  File "/home/work/.cache/huggingface/modules/transformers_modules/a5ba5501eb873d40d48bd0983bd2a8dd006bb838/quantization.py", line 128, in __init__
    assert str(weight.device).startswith('cuda'), 'The weights that need to be quantified should be on the CUDA device'
AssertionError: The weights that need to be quantified should be on the CUDA device

2.  使用 .cuda().quantize(4)   就是  **self.model = AutoModel.from_pretrained(MODEL_PATH, trust_remote_code=True, device_map="auto").cuda().quantize(4).eval()**  **显存还是12676M**
3.  使用 .quantize(4)     就是 **self.model = AutoModel.from_pretrained(MODEL_PATH, trust_remote_code=True, device_map="auto").quantize(4).eval()**      **显存还是12676M**

--- a/basic_demo/cli_demo.py
+++ b/basic_demo/cli_demo.py
@@ -1,11 +1,14 @@
 import os
 import platform
 from transformers import AutoTokenizer, AutoModel
+
 MODEL_PATH = os.environ.get('MODEL_PATH', 'THUDM/chatglm3-6b')
 TOKENIZER_PATH = os.environ.get("TOKENIZER_PATH", MODEL_PATH)
 
 tokenizer = AutoTokenizer.from_pretrained(TOKENIZER_PATH, trust_remote_code=True)
-model = AutoModel.from_pretrained(MODEL_PATH, trust_remote_code=True, device_map="auto").eval()
+model = AutoModel.from_pretrained(MODEL_PATH, trust_remote_code=True).eval()
+# add .quantize(bits=4, device="cuda").cuda() before .eval() to use int4 model
+# must use cuda to load int4 model
 
 os_name = platform.system()
 clear_command = 'cls' if os_name == 'Windows' else 'clear'

--- a/composite_demo/client.py
+++ b/composite_demo/client.py
@@ -138,7 +138,8 @@ class HFClient(Client):
                 trust_remote_code=True,
                 config=config,
                 device_map="auto").eval()
-            # add .quantize(4).cuda() before .eval() and remove device_map="auto" to use int4 model
+            # add .quantize(bits=4, device="cuda").cuda() before .eval() and remove device_map="auto" to use int4 model
+            # must use cuda to load int4 model
             prefix_state_dict = torch.load(os.path.join(pt_checkpoint, "pytorch_model.bin"))
             new_prefix_state_dict = {}
             for k, v in prefix_state_dict.items():
@@ -148,7 +149,8 @@ class HFClient(Client):
             self.model.transformer.prefix_encoder.load_state_dict(new_prefix_state_dict)
         else:
             self.model = AutoModel.from_pretrained(MODEL_PATH, trust_remote_code=True, device_map="auto").eval()
-            # add .quantize(4).cuda() before .eval() and remove device_map="auto" to use int4 model
+            # add .quantize(bits=4, device="cuda").cuda() before .eval() and remove device_map="auto" to use int4 model
+            # must use cuda to load int4 model
 
     def generate_stream(
             self,

